layouts/shortcodes/openstreetmap.html

```diff
@@ -0,0 +1,23 @@
+{{ $mapName := .Get "mapName" }}
+
+{{ $mapWidth := .Get "mapWidth" | default "100%" }}
+{{ $mapHeight := .Get "mapHeight" | default "600px" }}
+
+{{ $scaleControl := .Get "scaleControl" | default "true"  }}
+{{ $miniMap := .Get "miniMap" | default "false" }}
+{{ $scrollWheelZoom := .Get "scrollWheelZoom" | default "true" }}
+{{ $zoomControl := .Get "zoomControl" | default "true" }}
+{{ $allowEdit := .Get "allowEdit" | default "false" }}
+{{ $moreControl := .Get "moreControl" | default "true" }}
+{{ $searchControl := .Get "searchControl" | default "true" }}
+{{ $tilelayersControl := .Get "tilelayersControl" | default "null" }}
+{{ $embedControl := .Get "embedControl" | default "null" }}
+{{ $datalayersControl := .Get "datalayersControl" | default "true" }}
+{{ $onLoadPanel := .Get "onLoadPanel" | default "none" }}
+{{ $captionBar := .Get "captionBar" | default "false" }}
+
+{{ $scale := .Get "scale" }}
+{{ $coordX := .Get "coordX" }}
+{{ $coordY := .Get "coordY" }}
+
+<iframe width="{{ $mapWidth }}" height="{{ $mapHeight }}" frameBorder="0" src="https://umap.openstreetmap.fr/en/map/{{- $mapName -}}?scaleControl={{ $scaleControl }}&miniMap={{ $miniMap }}&scrollWheelZoom={{ $scrollWheelZoom }}&zoomControl={{ $zoomControl }}&allowEdit={{ $allowEdit }}&moreControl={{ $moreControl }}&searchControl={{ $searchControl }}&tilelayersControl={{ $tilelayersControl }}&embedControl={{ $embedControl }}&datalayersControl={{ $datalayersControl }}&onLoadPanel={{ $onLoadPanel }}&captionBar={{ $captionBar }}{{ with $scale}}#{{ . }}{{ end }}/{{ $coordX }}/{{ $coordY }}"></iframe>

```

Comment:
One request: Can we put `class="openstreetmap"` or something like that on this iframe? To make it easier to target with CSS.

Reply:
Also it looks like the `frameborder` attribute [is depreciated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/iframe#deprecated_attributes). If it's not too much to ask it'd be nice to add CSS to remove the border and remove this attribute. If that feels like more than you want to do we can always update it later (especially if we have that CSS class already there).

Reply:
I agree and while I was doing this, I thought how should css file be added in this case. Should it be added as a custom css as mentioned [here](https://github.com/chipzoller/hugo-clarity#custom-css-and-js) or is custom css a concept for users of this theme only? In this it's gonna be part of this theme by design rather than a `custom` implementation

Reply:
Custom CSS is just for end-users of the theme, so if we incorporate this shortcode then I think it makes sense to incorporate CSS for it as well. So if you just put a class on the iframe, you can then add CSS (technically Sass) for it to, I would guess, [this file](https://github.com/chipzoller/hugo-clarity/blob/master/assets/sass/_components.sass).